### DISCUSSION TITLE
Launching per-application configuration?

### DISCUSSION THREAD

**Discussion by MattDemers** [LizardByte/community#325] [Sunshine]
Hey all. I access Sunshine through both a Steam Link hardware and my Mac work laptop.

For the latter, I'd like to be able to load a different `.conf` when I load a specific Application; I'd like to make a clone of the "Desktop" Application, and then have it load `mac.conf` (in my Sunshine folder, along with the default `.conf`), and then have it undo when I quit.

This basically saves me from having to mess around with Autohotkey to switch back and forth between Windows and Mac layouts for my keyboard. The changes on `mac.conf` would be to switch some keys around. I'd also like to keep the default "Desktop" application the way it is, for use on my Steam Link (with a Windows layout).

From my understanding, I would be creating an Application and then using a DO command, and then another command for UNDO. I'm unsure of which ones I would use, and then which application I would be launching (since the default Desktop app seems to not have the path to mstsc.exe like Geforce Experience Now did).

Would my DO just be `sunshine [pathtoSunshine]/config/mac.conf` and then my UNDO be `sunshine [pathtoSunshine]/config/sunshine.conf`?

Any help would be appreciated.

**Comment by ReenigneArcher** [comment]
`mstsc.exe` was a hack for Moonlight to stream the desktop with GFE. Sunshine doesn't need that, since it just streams your desktop no matter what commands are there.

Running multiple instances of Sunshine is not supported, so no I would not put `sunshine` as a do/undo command.

You would need to create two (or more) different applications in sunshine, with scripts to do/undo whatever you want. What those commands would need to be, I have no idea.

**Reply by MattDemers** [reply]
>You would need to create two (or more) different applications in sunshine, with scripts to do/undo whatever you want. What those commands would need to be, I have no idea.

Maybe I could like... have the command be the initialization of the AutoHotkey Script, and the undo moving it back to the normal keymapping. I was just curious to know if I could do it at the Sunshine level, since the input config seemed possible.

[a comment by an automated account is not shown]
